examples/dgfip_c/enchain_static.c

```diff
@@ -0,0 +1,211 @@
+    IRDATA_reset_erreur(irdata);
+  }
+}
+
+void add_erreur(T_irdata *irdata, T_erreur *erreur, char *code)
+{
+  T_discord *new_discord = NULL;
+
+  if (irdata->tas_discord == 0) {
+    new_discord = (T_discord *)malloc(sizeof(T_discord));
+  } else {
+    new_discord = irdata->tas_discord;
+    irdata->tas_discord = new_discord->suivant;
+  }
+
+  add_erreur_code(erreur, code);
+
+  new_discord->erreur = erreur;
+  new_discord->suivant = 0;
+  *irdata->p_discord = new_discord;
+  irdata->p_discord = &new_discord->suivant;
+
+  if (strcmp(erreur->isisf, "O")) {
+    if ((erreur->type == ANOMALIE) && (++irdata->nb_bloquantes >= irdata->max_bloquantes)) {
+//      longjmp(irdata->jmp_bloq, 1);
+    }
+  }
+}
+
+void free_erreur()
+{
+}
+
+#else
+
+static T_discord *discords = 0;
+static T_discord *tas_discord = 0;
+static T_discord **p_discord = &discords;
+static jmp_buf jmp_bloq;
+
+static void init_erreur(void)
+{
+  *p_discord = tas_discord;
+  tas_discord = discords;
+  discords = 0;
+  p_discord = &discords;
+}
+
+void add_erreur(T_erreur *erreur, char *code)
+{
+  T_discord *new_discord = NULL;
+
+  if (tas_discord == 0) {
+    new_discord = (T_discord *)malloc(sizeof(T_discord));
+  } else {
+    new_discord = tas_discord;
+    tas_discord = new_discord->suivant;
+  }
+
+  add_erreur_code(erreur, code);
+
+  new_discord->erreur = erreur;
+  new_discord->suivant = 0;
+  *p_discord = new_discord;
+  p_discord = &new_discord->suivant;
+
+  if (strcmp(erreur->isisf, "O")) {
+    if (erreur->type == ANOMALIE) {
+//      longjmp(jmp_bloq, 1);
+    }
+  }
+}
+
+void free_erreur()
+{
+  T_discord *temp_discords = discords;
+  T_discord *dd = NULL;
+  char *debut = NULL;
+  int i = 0;
+
+  while (temp_discords != NULL) {
+    dd = temp_discords;
+    temp_discords = temp_discords->suivant;
+    if (dd->erreur->message != NULL) {
+      debut = strstr(dd->erreur->message, " ((");
+      if (debut != NULL) {
+        free(dd->erreur->message);
+      }
+      dd->erreur->message = NULL;
+    }
+  }
+}
+
+#endif /* FLG_MULTITHREAD */
+
+T_discord * no_error(T_irdata *irdata)
+{
+  return NULL;
+}
+
+T_discord * err_NEGATIF(T_irdata *irdata)
+{
+#ifdef FLG_MULTITHREAD
+  init_erreur(irdata);
+  if (setjmp(irdata->jmp_bloq) != 0) {
+    return irdata->discords;
+  }
+  add_erreur(irdata, &erreur_A000, NULL);
+  return irdata->discords;
+#else
+  init_erreur();
+  if (setjmp(jmp_bloq) != 0) {
+    return discords;
+  }
+  add_erreur(&erreur_A000, NULL);
+  return discords;
+#endif /* FLG_MULTITHREAD */
+}
+
+double floor_g(double a)
+{
+  if (fabs(a) <= LONG_MAX) {
+    return floor(a);
+  } else {
+    return a;
+  }
+}
+
+int multimax_def(int nbopd, char *var)
+{
+  for (int i = 0; i < nbopd; i++) {
+    if (var[i] == 1) return 1;
+  }
+  return 0;
+}
+
+double multimax(double nbopd, double *var)
+{
+  double s = 0.0;
+  for (int i = 0; i < (int)nbopd; i++) {
+    if (var[i] >= s) s = var[i];
+  }
+  return s;
+}
+
+int modulo_def(int a, int b)
+{
```

Comment:
There is no modulo currently supported by the rest of mlang?

Reply:
That function comes from the original DGFiP file. Not sure if it's actually needed.